spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/zuul/filters/route/support/AbstractRibbonCommand.java

```diff
@@ -32,39 +32,48 @@
 import com.netflix.hystrix.HystrixCommandKey;
 import com.netflix.hystrix.HystrixCommandProperties;
+import com.netflix.hystrix.HystrixCommandProperties.ExecutionIsolationStrategy;
 import com.netflix.zuul.constants.ZuulConstants;
 import com.netflix.zuul.context.RequestContext;
 
 /**
  * @author Spencer Gibb
  */
-public abstract class AbstractRibbonCommand<LBC extends AbstractLoadBalancerAwareClient<RQ, RS>, RQ extends ClientRequest, RS extends HttpResponse> extends HystrixCommand<ClientHttpResponse> implements
-		RibbonCommand {
+public abstract class AbstractRibbonCommand<LBC extends AbstractLoadBalancerAwareClient<RQ, RS>, RQ extends ClientRequest, RS extends HttpResponse>
+		extends HystrixCommand<ClientHttpResponse> implements RibbonCommand {
 
 	protected final LBC client;
 	protected RibbonCommandContext context;
 
-	public AbstractRibbonCommand(LBC client, RibbonCommandContext context) {
-		this("default", client, context);
+	public AbstractRibbonCommand(LBC client, RibbonCommandContext context,
+			ZuulProperties zuulProperties) {
+		this("default", client, context, zuulProperties);
 	}
 
-	public AbstractRibbonCommand(String commandKey, LBC client, RibbonCommandContext context) {
-		super(getSetter(commandKey));
+	public AbstractRibbonCommand(String commandKey, LBC client,
+			RibbonCommandContext context, ZuulProperties zuulProperties) {
+		super(getSetter(commandKey, zuulProperties));
 		this.client = client;
 		this.context = context;
 	}
 
-	protected static Setter getSetter(final String commandKey) {
+	protected static Setter getSetter(final String commandKey,
+			ZuulProperties zuulProperties) {
 
-		// we want to default to semaphore-isolation since this wraps
-		// 2 others commands that are already thread isolated
 		// @formatter:off
-		final String name = ZuulConstants.ZUUL_EUREKA + commandKey + ".semaphore.maxSemaphores";
-		final DynamicIntProperty value = DynamicPropertyFactory.getInstance()
-				.getIntProperty(name, 100);
-		final HystrixCommandProperties.Setter setter = HystrixCommandProperties .Setter()
-				.withExecutionIsolationStrategy(HystrixCommandProperties.ExecutionIsolationStrategy.SEMAPHORE)
-				.withExecutionIsolationSemaphoreMaxConcurrentRequests(value.get());
+		final HystrixCommandProperties.Setter setter = HystrixCommandProperties.Setter()
+				.withExecutionIsolationStrategy(zuulProperties.getRibbonIsolationStrategy());
+		if (zuulProperties.getRibbonIsolationStrategy() == ExecutionIsolationStrategy.SEMAPHORE){
+			final String name = ZuulConstants.ZUUL_EUREKA + commandKey + ".semaphore.maxSemaphores";
+			// we want to default to semaphore-isolation since this wraps
+			// 2 others commands that are already thread isolated
+			final DynamicIntProperty value = DynamicPropertyFactory.getInstance()
+					.getIntProperty(name, 100);
+			setter.withExecutionIsolationSemaphoreMaxConcurrentRequests(value.get());
+		} else	{
+			// TODO Find out is some parameters can be set here
```

Comment:
For now I do not what parameters would be useful here so I left is open for improvement